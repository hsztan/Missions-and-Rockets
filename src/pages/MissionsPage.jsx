import MissionsList from '../components/MissionsList/MissionsList';

const MissionsPage = () => (
  <div>
    <MissionsList />
  </div>
);

export default MissionsPage;
