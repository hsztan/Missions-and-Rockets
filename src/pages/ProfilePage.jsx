import JoinedList from '../components/JoinedList/JoinedList';

const ProfilePage = () => (
  <div className="profile">
    <JoinedList />
  </div>
);

export default ProfilePage;
