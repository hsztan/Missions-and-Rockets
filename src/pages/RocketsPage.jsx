import RocketsList from '../components/RocketsList/RocketsList';

const RocketsPage = () => (
  <div>
    <RocketsList />
  </div>
);

export default RocketsPage;
