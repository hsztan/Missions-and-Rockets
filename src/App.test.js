import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
// import { fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from './redux/configure-store';

import MissionsPage from './pages/MissionsPage';
import RocketsPage from './pages/RocketsPage';
import ProfilePage from './pages/ProfilePage';

describe('Pages renders correctly', () => {
  it('Missions page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionsPage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <MissionsPage />
      </Provider>,
    );
    const element = document.querySelector('.missions-list');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
  it('Rockets page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketsPage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <RocketsPage />
      </Provider>,
    );
    const element = document.querySelector('.rockets-list');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
  it('Profile page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProfilePage />
        </Provider>,
      )
      .toJSON();
    render(
      <Provider store={store}>
        <ProfilePage />
      </Provider>,
    );
    const element = document.querySelector('.profile');
    expect(element).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
