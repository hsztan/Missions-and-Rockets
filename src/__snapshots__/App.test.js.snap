// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly Calculator page renders correctly 1`] = `
<div>
  <ul
    className="rockets-list"
  />
</div>
`;

exports[`Pages renders correctly Home page renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="joined"
  >
    <div
      className="missions"
    >
      <h1>
        My Missions
      </h1>
      <ul />
    </div>
    <div
      className="rockets"
    >
      <h1>
        My Rockets
      </h1>
      <ul />
    </div>
  </div>
</div>
`;

exports[`Pages renders correctly Missions page renders correctly 1`] = `
<div>
  <div
    className="missions-list"
  >
    <table>
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Pages renders correctly Profile page renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="joined"
  >
    <div
      className="missions"
    >
      <h1>
        My Missions
      </h1>
      <ul />
    </div>
    <div
      className="rockets"
    >
      <h1>
        My Rockets
      </h1>
      <ul />
    </div>
  </div>
</div>
`;

exports[`Pages renders correctly Quote page renders correctly 1`] = `
<div>
  <div
    className="missions-list"
  >
    <table>
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Pages renders correctly Rockets page renders correctly 1`] = `
<div>
  <ul
    className="rockets-list"
  />
</div>
`;
